core/app/helpers/spree/taxons_helper.rb

```diff
@@ -4,16 +4,26 @@ module TaxonsHelper
     # that we can use configurations as well as make it easier for end users to override this determination.  One idea is
     # to show the most popular products for a particular taxon (that is an exercise left to the developer.)
     def taxon_preview(taxon, max=4)
-      products = taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").limit(max)
+      products = taxon.active_products.select(*selection_sql).limit(max)
       if (products.size < max)
         products_arel = Spree::Product.arel_table
         taxon.descendants.each do |taxon|
           to_get = max - products.length
-          products += taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
+          products += taxon.active_products.select(*selection_sql).where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
           break if products.size >= max
         end
       end
       products
     end
+
+    private
+
+    def selection_sql
+      [
+        "DISTINCT (spree_products.id)",
```

Comment:
Using `DISTINCT` that way, will not work in PostgreSQL (if I undestand correctly, why distinct is used there). In [documentation](http://www.postgresql.org/docs/9.0/static/sql-select.html), we have example:

~~~ SQL
SELECT DISTINCT ON (location) location, time, report
    FROM weather_reports
    ORDER BY location, time DESC;
~~~

So when we want to show records with unique `spree_product.id` then we need to add `ON` before field name, but this is not compatible with other DBs.